logging/redaction/log_redaction_final.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import json
+import logging
+from typing import List
+
+from apache_beam import CombineFn, CombineGlobally, DoFn, io, ParDo, Pipeline, WindowInto
```

Comment:
ditto.

Reply:
Many of the above are classes (e.g. `ParDo` or `List`).
Also the code following the [public documentation][1] with the example from [this repo][2] that uses this approach. 

[1]: https://cloud.google.com/pubsub/docs/stream-messages-dataflow#python
[2]: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/HEAD/pubsub/streaming-analytics/PubSubToGCS.py